andela/ah-the-jedi-backend
# #165305284 Users should be able to see their reading stats

### What does this PR do?
Implements functionality to track article reading statistics

### Description of Task(s) to be completed?
- ensure read count is logged when an authenticated user reads an article
- ensure only authenticated users can see article read stats (read count)
- ensure article authors can see the people who have read their articles

### How should this be manually tested?

Clone this repository on local machine and CD into project folder
`$ git clone git@github.com:andela/ah-the-jedi-backend.git`
`$ cd ah-the-jedi-backend`

Check out to the PR's branch
`$ git checkout ft-reading-stats-165305284`

Create and activate virtual environment
`$ python3 -m venv venv`
`$ source .venv/bin/activate`

Install app dependencies from requirements.txt
`$ pip install -r api/requirements.txt`

Make migrations and run the server
`$ python manage.py makemigrations`
`$ python manage.py migrate`
`$ python manage.py runserver`

**Use postman to test the endpoints**


           POST <your-localhost>/api/users/



       POST <your-localhost>/api/login/ 

       Use the token provided after login for authorization 

- [x] **create an article**

      `POST`  `<you-local-host>/api/articles/`

- [x] **get the created article and include authentication details**

      `GET`  `<you-local-host>/api/articles/<slug>/`

### What are the relevant PT stories?
[#165305284](https://www.pivotaltracker.com/story/show/165305284)

### Screenshots
#### Authenticated user and article owner
![image](https://user-images.githubusercontent.com/11965186/57854337-b2f88a80-77f0-11e9-9dd6-c08d965901f5.png)

#### Authenticated user but not the author
![image](https://user-images.githubusercontent.com/11965186/57854391-ce639580-77f0-11e9-8f6e-0d21604f3f05.png)

#### Unauthenticated user
![image](https://user-images.githubusercontent.com/11965186/57854430-e6d3b000-77f0-11e9-839d-a030480620ef.png)
